sttz.InstallUnity/Installer/Platforms/WIndowsPlatform.cs

```diff
@@ -0,0 +1,316 @@
+            }
+
+            var installPath = GetInstallationPath(installing.version, installationPaths);
+            // TODO: start info runas
+            var result = await RunAsAdmin(item.filePath, $"/S /D={installPath}");
+            if (result.exitCode != 0)
+            {
+                throw new Exception($"Failed to install {item.filePath} output: {result.output} / {result.error}");
+            }
+
+            if (item.package.name == PackageMetadata.EDITOR_PACKAGE_NAME)
+            {
+                installedEditor = true;
+            }
+        }
+
+        public async Task MoveInstallation(Installation installation, string newPath, CancellationToken cancellation = default)
+        {
+            // do nothing
+        }
+
+        public async Task PrepareInstall(UnityInstaller.Queue queue, string installationPaths, CancellationToken cancellation = default)
+        {
+            if (installing.version.IsValid)
+                throw new InvalidOperationException($"Already installing another version: {installing.version}");
+
+            installing = queue.metadata;
+            this.installationPaths = installationPaths;
+            installedEditor = false;
+
+            // Check for upgrading installation
+            if (!queue.items.Any(i => i.package.name == PackageMetadata.EDITOR_PACKAGE_NAME))
+            {
+                var installs = await FindInstallations(cancellation);
+                var existingInstall = installs.Where(i => i.version == queue.metadata.version).FirstOrDefault();
+                if (existingInstall == null)
+                {
+                    throw new InvalidOperationException($"Not installing editor but version {queue.metadata.version} not already installed.");
+                }
+
+                installedEditor = true;
+            }
+        }
+
+        public async Task<bool> PromptForPasswordIfNecessary(CancellationToken cancellation = default)
+        {
+            // Don't care about password. The system will ask for elevated priviliges automatically
+            return true;
+        }
+
+        public async Task Uninstall(Installation installation, CancellationToken cancellation = default)
+        {
+            var result = await RunAsAdmin(Path.Combine(installation.path, "Editor", "Uninstall.exe"), "/AllUsers /Q /S");
+            if (result.exitCode != 0)
+            {
+                throw new Exception($"Could not uninstall Unity. output: {result.output}, error: {result.error}");
+            }
+
+            Logger.LogDebug($"Unity {installation.version} uninstalled successfully");
+
+            try
+            {
+                // TODO: Should folder be deleted even when uninstall command returns with exitcode != 0?
+                Logger.LogInformation($"Deleting folder path {installation.path} recursively");
+                await Task.Delay(1000); // Wait for uninstallation
```

Comment:
Add a why comment: 

// Sometimes the files are still locked even after the uninstall process has exited. Waiting a second seems to help.